### PR TITLE
decouple exists and contains from observable

### PR DESCRIPTION
**Pull request by gchudnov**
Decoupled [exists and contains](http://reactivex.io/documentation/operators/contains.html) from observable.
Please, review

Thank you,
Grigoriy

### CODE REVIEW

File: Rx/v2/src/rxcpp/operators/rx-any.hpp [hunk 1]
```diff
@@ -78,37 +113,99 @@ struct any
...
+template<class... AN>
+auto contains(AN&&... an)
+->     operator_factory<contains_tag, AN...> {
+    return operator_factory<contains_tag, AN...>(std::make_tuple(std::forward<AN>(an)...));
 }
+
 }
 
+template<>
+struct member_overload<exists_tag>
```
Comment (kirkshoop) [inline Rx/v2/src/rxcpp/operators/rx-any.hpp]: I do not see any specialization for `any_tag`, did I miss it?
Do we have any tests for `any()` or only `exists()` and `contains()`?

Comment (kirkshoop) [inline Rx/v2/src/rxcpp/operators/rx-any.hpp]: Looks Great!

Perhaps we can just change one of the `exists()` test to use `any()` to ensure that it does not get broken later.

**Comment by kirkshoop** [comment]
Thank You!

File: Rx/v2/src/rxcpp/operators/rx-any.hpp (hunk 1 above)
Comment (gchudnov) [inline Rx/v2/src/rxcpp/operators/rx-any.hpp]: Sure, I'll do in in one of the next PRs

**Pull request merged**